code/modules/research/xenoarchaeology/geosample.dm

```diff
@@ -34,6 +34,18 @@
 	pixel_x = rand(-8, 8) * PIXEL_MULTIPLIER
 	pixel_y = rand(-8, 0) * PIXEL_MULTIPLIER
 
+/obj/item/weapon/rocksliver/Destroy()
+	geological_data = null
+	..()
+
+/obj/item/weapon/rocksliver/throw_impact(atom/hit_atom)
+	var/turf/T = get_turf(src)
+	if (T && !istype(T, /turf/space))
+		visible_message("<span class='danger'>The virus dish shatters on impact!</span>")
```

Comment:
This is some weird tasting pasta chef.

Reply:
More like this is what 4AM code tastes like.